Zend/zend_API.c

```diff
@@ -1682,7 +1682,12 @@ ZEND_API int array_set_zval_key(HashTable *ht, zval *key, zval *value) /* {{{ */
 			result = zend_hash_index_update(ht, Z_LVAL_P(key), value);
 			break;
 		case IS_DOUBLE:
-			result = zend_hash_index_update(ht, zend_dval_to_lval(Z_DVAL_P(key)), value);
+			if (Z_DVAL_P(key) <= INT_MAX) {
```

Comment:
Is it intentional to ignore out-of-bounds negatives? Otherwise, there should also be a `Z_DVAL_P(key) >= INT_MIN` check.